log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -405,6 +405,7 @@ public boolean stop(final long timeout, final TimeUnit timeUnit) {
             }
         }
         LOGGER.debug("Stopped LoggerContext[name={}, {}] with status {}", getName(), this, true);
+    	StatusLogger.getLogger().prepareToStop();
```

Comment:
Again, I think I am missing some context here. Would you mind elaborating a bit on the necessity of this line here, please?

Reply:
Something needs to tell the StatusLogger there is a stop event. Otherwise it never releases the file handles for the listeners writing to a file. 

In the code on master on windows the status logger log file is never closed so the file remains locked which causes the test to fail. Now the LoggerContext is telling the StatusLogger that a shutdown event is happening. In the code I'll push later, I've added a paremeter to say which context is shutting down so the StatusLogger can remove the listeners for that context.